cpp/src/quantiles/tdigest/tdigest_aggregation.cu

```diff
@@ -623,17 +630,18 @@ std::unique_ptr<column> build_output_column(size_type num_rows,
 
   // adjust offsets.
   rmm::device_uvector<size_type> sizes(num_rows, stream);
-  thrust::transform(rmm::exec_policy(stream),
-                    thrust::make_counting_iterator(0),
-                    thrust::make_counting_iterator(0) + num_rows,
-                    sizes.begin(),
-                    [offsets = offsets->view().begin<size_type>()] __device__(size_type i) {
-                      return offsets[i + 1] - offsets[i];
-                    });
+  auto offsets_iter = offsets->view().begin<size_type>();
+  thrust::adjacent_difference(rmm::exec_policy(stream),
+                              offsets->view().begin<size_type>() + 1,
+                              offsets->view().end<size_type>(),
+                              sizes.begin());
```

Comment:
This rewrite assumes that the first offset value is always 0 while the original transform code does not.